geocity/apps/submissions/templates/submissions/_submission_summary.html

```diff
@@ -116,7 +116,7 @@ <h2>Documents</h2>
                 </div>
                 <div class="col-2 text-right">
                   <a class="align-middle" href="{% url 'submissions:complementary_documents_download' document.name %}"><i class="fa fa-download" aria-hidden="true"></i></a>
-                  {% if user == document.owner or user.is_superuser %}
+                  {% if user == document.owner and user != author or user.is_superuser %}
                     <form class="d-inline-block" method="post" action="{% url 'submissions:complementary_documents_delete' pk=document.id %}">
```

Comment:
The `complementary_documents_delete` must also be protected. Ask @AlexandreJunod if needed